File: crates/nil-payload/src/cheat/player.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::player::PlayerId;
use nil_core::world::config::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatGetPlayerRequest {
  pub world: WorldId,
  #[serde(default)]
  pub player: Option<PlayerId>,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatGetPlayersRequest {
  pub world: WorldId,
}

File: app/src/types/core/player.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { Resources } from '@/types/core/resources';

export interface PublicPlayer {
  readonly id: PlayerId;
  readonly status: PlayerStatus;
}

export interface Player extends PublicPlayer {
  readonly resources: Resources;
}

export type PlayerId = string;

export type PlayerStatus = 'active' | 'inactive';

export interface PlayerOptions {
  readonly id: PlayerId;
}
